py/server/deephaven/server/plugin/object/__init__.py

```diff
@@ -27,28 +28,45 @@ class ExporterAdapter(Exporter):
     def __init__(self, exporter: _JExporterAdapter):
         self._exporter = exporter
 
-    def reference(self, object, allow_unknown_type : bool = False, force_new : bool = False) -> Optional[Reference]:
-        object = _unwrap(object)
-        if isinstance(object, jpy.JType):
-            ref = self._exporter.reference(object, allow_unknown_type, force_new)
+    def reference(self, obj: Any, allow_unknown_type: bool = True, force_new: bool = True) -> Optional[Reference]:
+        obj = _unwrap(obj)
+        if isinstance(obj, jpy.JType):
+            ref = self._exporter.reference(obj, allow_unknown_type, force_new)
         else:
-            ref = self._exporter.referencePyObject(object, allow_unknown_type, force_new)
+            ref = self._exporter.referencePyObject(obj, allow_unknown_type, force_new)
         return _adapt_reference(ref) if ref else None
 
     def __str__(self):
         return str(self._exporter)
 
 
+class MessageStreamAdapter(MessageStream):
+    def __init__(self, wrapped: _JMessageStream):
+        self._wrapped = wrapped
+
+    def on_data(self, payload: bytes, references: List[Any]):
+        self._wrapped.on_data(payload, [_unwrap(ref) for ref in references])
+
+    def on_close(self):
+        self._wrapped.on_close()
+
+
 # see io.deephaven.server.plugin.python.ObjectTypeAdapter for calling details
 class ObjectTypeAdapter:
     def __init__(self, user_object_type: ObjectType):
         self._user_object_type = user_object_type
 
-    def is_type(self, object):
-        return self._user_object_type.is_type(object)
+    def is_type(self, obj):
+        return self._user_object_type.is_type(obj)
+
+    def is_fetch_only(self):
+        return isinstance(self._user_object_type, FetchOnlyObjectType)
+
+    def to_bytes(self, exporter: _JExporterAdapter, obj: Any):
+        return self._user_object_type.to_bytes(ExporterAdapter(exporter), obj)
 
-    def to_bytes(self, exporter: _JExporterAdapter, object):
-        return self._user_object_type.to_bytes(ExporterAdapter(exporter), object)
+    def create_client_connection(self, obj: Any, connection: _JMessageStream):
```

Comment:
A private type is included in a public API.

@jmao-denver

Reply:
This is not a public API - the public part can be found in https://github.com/deephaven/deephaven-plugin/pull/12. This is the adapter that creates the user-defined plugin, which implements the api linked in that other PR.